File: C++/Mainoverloading.cpp
#include <iostream>
using namespace std;

// Standard entry-point main function
int main() {
    cout << "Standard main function called." << endl;

    // Call the overloaded main functions
    main(10);           // Calls the version with one int parameter
    main(10, 20);       // Calls the version with two int parameters

    return 0;
}

// Overloaded main function with one parameter
void main(int x) {
    cout << "Overloaded main function with one parameter called. x = " << x << endl;
}
// Overloaded main function with two parameters
void main(int x, int y) {
    cout << "Overloaded main function with two parameters called. x = " << x << ", y = " << y << endl;
}

File: Mainoverloading.java
class MainOverloadExample {

    // Standard main method called by JVM
    public static void main(String[] args) {
        System.out.println("Standard main method");

        // Calling overloaded main methods explicitly
        main(10);
        main("Hello");
        main(5.5, 2.2);
    }

    // Overloaded main method with an int parameter
    public static void main(int a) {
        System.out.println("Overloaded main with int: " + a);
    }

    // Overloaded main method with a String parameter
    public static void main(String str) {
        System.out.println("Overloaded main with String: " + str);
    }

    // Overloaded main method with two double parameters
    public static void main(double a, double b) {
        System.out.println("Overloaded main with doubles: " + (a + b));
    }
}
